python/cudf_polars/cudf_polars/experimental/io.py

```diff
@@ -47,3 +50,219 @@ def _(
+        skip_rows: int,
+        n_rows: int,
+        row_index: tuple[str, int] | None,
+        predicate: NamedExpr | None,
+    ):
+        """Evaluate and return a dataframe."""
+        if typ not in ("parquet",):  # pragma: no cover
+            raise NotImplementedError(f"Unhandled Scan type for file splitting: {typ}")
+
+        rowgroup_metadata = plc.io.parquet_metadata.read_parquet_metadata(
+            plc.io.SourceInfo(paths)
+        ).rowgroup_metadata()
+        total_row_groups = len(rowgroup_metadata)
+        if total_splits > total_row_groups:
+            # Don't bother aligning on row-groups
+            total_rows = sum(rg["num_rows"] for rg in rowgroup_metadata)
+            n_rows = int(total_rows / total_splits)
+            skip_rows = n_rows * split_index
+        else:
+            # Align split with row-groups
+            rg_stride = int(total_row_groups / total_splits)
+            skip_rgs = rg_stride * split_index
+            skip_rows = (
+                sum(rg["num_rows"] for rg in rowgroup_metadata[:skip_rgs])
+                if skip_rgs
+                else 0
+            )
+            n_rows = sum(
+                rg["num_rows"]
+                for rg in rowgroup_metadata[skip_rgs : skip_rgs + rg_stride]
+            )
+
+        # Last split should always read to end of file
+        if split_index == (total_splits - 1):
+            n_rows = -1
+
+        return Scan.do_evaluate(
+            schema,
+            typ,
+            reader_options,
+            config_options,
+            paths,
+            with_columns,
+            skip_rows,
+            n_rows,
+            row_index,
+            predicate,
+        )
+
+
+def _sample_pq_statistics(ir: Scan) -> dict[str, float]:
+    import numpy as np
+    import pyarrow.dataset as pa_ds
+
+    # Use average total_uncompressed_size of three files
+    # TODO: Use plc.io.parquet_metadata.read_parquet_metadata
+    n_sample = 3
+    column_sizes = {}
+    ds = pa_ds.dataset(ir.paths[:n_sample], format="parquet")
+    for i, frag in enumerate(ds.get_fragments()):
+        md = frag.metadata
+        for rg in range(md.num_row_groups):
+            row_group = md.row_group(rg)
+            for col in range(row_group.num_columns):
+                column = row_group.column(col)
+                name = column.path_in_schema
+                if name not in column_sizes:
+                    column_sizes[name] = np.zeros(n_sample, dtype="int64")
+                column_sizes[name][i] += column.total_uncompressed_size
+
+    return {name: np.mean(sizes) for name, sizes in column_sizes.items()}
+
+
+def _scan_partitioning(ir: Scan) -> tuple[int, int]:
+    split, stride = 1, 1
+    if ir.typ == "parquet":
+        file_size: float = 0
+        # TODO: Use system info to set default blocksize
+        parallel_options = ir.config_options.get("executor_options", {})
+        blocksize: int = parallel_options.get("parquet_blocksize", 1024**3)
```

Comment:
>Is 1GiB a good size, or should we pick something larger?

My experience tells me that 1GiB is a good default, but that most users with datacenter-class GPUs will usually want to go bigger. In Dask cuDF we use pynvml to query the "real" device size. The details of this can get sticky, so I'd rather revisit this kind of improvement after we start benchmarking.